packages/dht/src/dht/store/StoreRpcRemote.ts

```diff
@@ -20,10 +20,10 @@ export class StoreRpcRemote extends RpcRemote<IStoreRpcClient> {
         }
     }
 
-    async replicateData(request: ReplicateDataRequest, doNotConnect: boolean = false): Promise<void> {
+    async replicateData(request: ReplicateDataRequest): Promise<void> {
         const options = this.formDhtRpcOptions({
             timeout: EXISTING_CONNECTION_TIMEOUT,
-            doNotConnect
+            doNotConnect: false
```

Comment:
`doNotConnect` should be set true atleast for when calling `this.replicateDataToClosestNodes()` from the destroy function. This way the node will not open any new soon to be disconnected connections during tear down.

Setting `doNotConnect: true` makes sense in general as all nodes in the kBucket structure should be connected to.

The only exception would be the replication that is done on new contacts, as the node might not be connected to yet.